### PR TITLE
adding batch support

### PR DESCRIPTION
**Pull request by joe-ayoub-segment**
Adding batch capability to Optimizely Audiences. 

## Testing

Tested locally.

### CODE REVIEW

**Review by sayan-das-in — APPROVED** [review]

**Review by Romani-Ramzi — APPROVED** [review]

**Review by Romani-Ramzi — APPROVED** [review]

**Comment by Romani-Ramzi** [comment]
@joe-ayoub-segment seems good, tests are updated to test either flow, the batch and the single perform method.
All good from our side

**Comment by joe-ayoub-segment** [comment]
Note: Can't deploy until tested in Staging.

**Review by varadarajan-tw — APPROVED** [review]

File: packages/destination-actions/src/destinations/optimizely-advanced-audience-targeting/syncAudience/optimizely-client.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  request: RequestClient
+  payloads: Payload[]
+  settings: Settings
+  propsOrTraits: PropsOrTraits[]
+
+  constructor(request: RequestClient, data: Data) {
+    this.request = request
+    this.settings = data.settings
+    this.payloads = Array.isArray(data.payload) ? data.payload : [data.payload]
+    this.propsOrTraits = Array.isArray(data.rawData) ? data.rawData : [data.rawData]
```
Comment (varadarajan-tw) [inline packages/destination-actions/src/destinations/optimizely-advanced-audience-targeting/syncAudience/optimizely-client.ts:27]: Why are we using rawData here?

Comment (varadarajan-tw) [reply packages/destination-actions/src/destinations/optimizely-advanced-audience-targeting/syncAudience/optimizely-client.ts:27]: could we just add a mapping with $.properties or $.traits and use that instead.

**Comment by joe-ayoub-segment** [comment]
hi @Romani-Ramzi this PR has been deployed

**Comment by Romani-Ramzi** [comment]
will "perform" function still send single data but in array style?
does it mean that we will still keep getting high traffic but its just an array?

**Pull request merged**